### DISCUSSION TITLE
Support for streams

### DISCUSSION THREAD

**Discussion by mgrishaber-merlin** [photostructure/exiftool-vendored.js#180] [Q&A] [answered]
Hello,
I am wondering if there is support for streams.  I have a scenario where I would like to use the `rewriteAllTags` function to repair exif tags on an incoming stream and pipe that stream to ImageMagick to further convert the image.
From what I can tell I will have to write the incoming stream to a temp file, call `rewriteAllTags` on that file and then create a read stream for the resulting temp file.
Is there a way to avoid creating temp files and use the stream directly, and is there any problem with performing operations concurrently?

Thanks in advance

**Comment by mceachen** [comment, marked as answer]
See https://github.com/photostructure/exiftool-vendored.js/issues/2 and https://github.com/photostructure/exiftool-vendored.js/issues/100
